src/ActionJob.php

```diff
@@ -21,13 +23,20 @@ class ActionJob implements ShouldQueue
     /** @var array */
     protected $tags = ['action_job'];
 
+    /** @var callable */
+    protected $onFailCallback;
+
     public function __construct($action, array $parameters = [])
     {
         $this->actionClass = is_string($action) ? $action : get_class($action);
         $this->parameters = $parameters;
 
         if (is_object($action)) {
             $this->tags = $action->tags();
+
+            if (method_exists($action, 'failed')) {
+                $this->onFailCallback = [$action, 'failed'];
```

Comment:
@freekmurze @brendt I'm not too knowledgeable about queues and serializing. I'd appreciate some input on this.

Originally, I wanted to set this property to a closure like:
```php
function (Exception $exception) use ($action) {
    return $action->failed($exception);
}
```

However, that resulted in an error saying that Closures can't be serialized.

So I went with this callable notation, but I'm not 100% sure that this is correct either.

What if `$action` is some unserializable object? Or an object that injects unserializable classes (like PDO connection instances, for example).

If that's an issue, then feel free to close this PR as I don't see any other way to achieve this. Either we need to store a Closure (which is impossible to serialize), or we need to store a reference to an object's method - and as an extension, store the object itself (which *might* be impossible to serialize?)

Or maybe we can just add a notice that the method works only with serializable actions.

Reply:
> What if $action is some unserializable object? Or an object that injects unserializable classes (like PDO connection instances, for example).

It's a known constraint, there's little we can do about that. The same would happen if you'd try to store a closure or exception or something else unserializable on a job. Nothing we can do about that. So I'd say it shouldn't be a concern. I think it's a good solution!